pkg/utils/limited_backoff_test.go

```diff
@@ -0,0 +1,75 @@
+package utils
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/cenkalti/backoff"
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestLimitedBackOff(t *testing.T) {
+	Convey("Subject: Using limited backoff", t, func() {
+		const (
+			interval   = time.Second
+			retryLimit = 3
+		)
+
+		Convey("Given a new limited backoff", func() {
+			backOff := NewLimitedBackOff(interval, retryLimit)
+
+			testLimitedBackOff(backOff, retryLimit, interval)
+		})
+
+		Convey("Given an exhausted limited backoff", func() {
+			backOff := NewLimitedBackOff(interval, retryLimit)
+			callMultipleNextBackOffs(backOff, retryLimit)
+
+			Convey("When calling Reset", func() {
+				backOff.Reset()
+
+				assertRetryCount(backOff, 0)
+
+				testLimitedBackOff(backOff, retryLimit, interval)
+			})
+		})
+	})
+}
+
+func testLimitedBackOff(backOff *limitedBackOff, retryLimit int, interval time.Duration) {
```

Comment:
Not a blocker at all but this seems just a touch more complicated than it needs to be for the class you're testing.